### DISCUSSION TITLE
When loading from SharePoint, show SharePoint list in alphabetical order.

### DISCUSSION THREAD

**Discussion by Tom-Enns** [jgraph/drawio#2870] [Ideas]
When I load a diagram from SharePoint the list of SharePoint sites I have access to is loaded in what appears to be a random order. I would love if this list was displayed in alphabetical order.

**Comment by snydergd** [comment]
I have the exact same problem using the [Microsoft Team's integration](https://appsource.microsoft.com/en-US/product/office/WA200003444?tab=Overview) - trying to add a tab for an existing file.  This is very important to me because there are a large number of SharePoint locations listed and it is extremely difficult to find the one I am looking for.  In my case it actually prevents me from being able to use the (excellent) integration at all.

![Screenshot of problematic viewer](https://github.com/jgraph/drawio/assets/1040083/4a68c6db-a268-45bb-9771-5ceb65a35308)

**Reply by snydergd** [reply]
@davidjgraph Thanks for the quick reply!  I can follow up with Microsoft on it, then.  When I go to add a tab in teams, a dialog selects the app (e.g., draw.io), then each app seems to have its own unique selection process for selecting or creating the content the app should deal with in that tab.  Draw.io shows differently than Word, Excel, etc.  So it seems like the application has some control over what Microsoft displays at this point.

Do you happen to know what Microsoft function is being used here by the app to tell Microsoft "have the user pick a file"?  Or is it just some sort of default that apps can use?  If the source is available for this integration I can go look at that, too.  I haven't been able to find it yet.

**Reply by snydergd** [reply]
After a little more investigation, it looks like the way this screen works is that it's an iFrame served from `https://app.diagrams.net/connect/teams/tasks/odPicker.html?&teamName=myteamname&channelRelativeUrl=%2Fsites%2Fmyteamname%2FShared%20Documents%2Fmychannel&teamSiteUrl=https%3A%2F%2Fmysite.sharepoint.com%2Fsites%2Fmyteam&channelName=mychannel`.  The frame appears to be using the Microsoft Graph [sites search endpoint](https://learn.microsoft.com/en-us/graph/api/site-search?view=graph-rest-1.0&tabs=http) to populate the list.

In my case it doesn't seem to be giving back my team/site at all in the list, but interestingly, going after the main sites graph endpoint (`/sites`) gives an empty list and `/sites/getAllSites` endpoint says access denied.  The token used has the scopes `Files.ReadWrite.All Sites.Read.All User.Read profile openid email`.  The response to the search is 136 items like in the list, so it doesn't seem like a paging issue.  That's all probably a separate issue, though.

In theory it seems like you should be able to use common query string parameters to control the sort order of this response, [like described here](https://learn.microsoft.com/en-us/graph/query-parameters?tabs=http), but I haven't had much luck getting that to work.

Maybe this page could instead just sort the results with JavaScript.  It doesn't seem like there is any paging going on that would be an obstacle either.

As a workaround in my case, I have used the browser version of Teams, inspected that frame, and changed the URL to point to the base of my Team's channel.  This way I can go to the "Shared" section and choose files directly from the channel's base file location, without having to use the sharepoint site list at all.

**Reply by gdsnyderfedins** [reply]
Linking a few related posts:
- https://github.com/jgraph/drawio/issues/4382
- https://github.com/jgraph/drawio/issues/4188
- https://github.com/jgraph/drawio/issues/3514

I still think that a search or sort would be extremely helpful, especially within the teams application.  However, in the web version in a browser it looks like using the browser's search (Ctrl-F) seems to work in the meantime.

For my issue of sites being missing, it seems that the search endpoint only surfaces "public" sites - ones that show up in the Sharepoint search.  For example, I have created a shared channel in Microsoft Teams and it shows up in the search and in drawio.  It also shows sites that were created as public sites (another team creates the sites, so I am not sure what "public" means - in searching it looks like maybe private ones are explicitly blacklisted for search: https://techcommunity.microsoft.com/discussions/sharepoint_general/sharepoint-online-search-will-not-find-some-sites-that-i-have-created/3884155)  I would have to confirm - in the meantime posting in case someone else can check this.
